### PR TITLE
Add support for CAF/Alac audio

### LINKED ISSUE #5: Support ALAC format decoding via `alac` crate?
Here's the [alac](https://crates.io/crates/alac) crate.

### PR DESCRIPTION
**Pull request by est31**
Adds support for the Alac codec, right now only inside the CAF container.

Also, ALAC is the only format in CAF that's supported currently.

Fixes #5.

r? @mitchmindtree

### CODE REVIEW

**Comment by est31** [comment]
ping @mitchmindtree

**Comment by mitchmindtree** [comment]
Ahh I forgot about this, thanks for the reminder @est31, will look at this asap :+1:

File: README.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ Supported Formats
 | FLAC | "flac" | YES | - | flac | [claxon](https://crates.io/crates/claxon) |
 | Ogg Vorbis | "ogg", "oga" | YES | - | ogg_vorbis | [lewton](https://crates.io/crates/lewton) |
 | WAV | "wav", "wave" | YES | - | wav | [hound](https://crates.io/crates/hound) |
+| ALAC (within CAF) | YES | - | caf_alac | [caf](https://crates.io/crates/caf) [alac](https://crates.io/crates/alac) |
```
Comment (mitchmindtree) [inline README.md]: Looks like the table is missing the extensions column:

screen shot 2017-02-08 at 10 38 53 pm

File: src/lib.rs [hunk 2]
```diff
@@ -22,6 +29,8 @@ pub enum Format {
     OggVorbis,
     #[cfg(feature="wav")]
     Wav,
+    #[cfg(feature="caf")]
```
Comment (mitchmindtree) [inline src/lib.rs]: Should this be the `caf_alac` feature?

File: src/lib.rs [hunk 3]
```diff
@@ -51,6 +62,8 @@ impl Format {
             Format::Wav => "wav",
             #[cfg(feature="ogg_vorbis")]
             Format::OggVorbis => "ogg",
+            #[cfg(feature="caf")]
```
Comment (mitchmindtree) [inline src/lib.rs]: and this

**Review by mitchmindtree — CHANGES_REQUESTED** [review]
Just a few minor things, otherwise looks great! Nice one :)

Oh also, it would be great if you got a chance to add it to the chain in the `sine.rs` example, but probs not totally necessary.

**Comment by mitchmindtree** [comment]
Oh also, would you mind adding a line to the `.travis.yml` script section that *just* tests the `caf_alac` feature? I think it's useful to have travis test each feature individually as it is easy to forget to add a missing feature-gate, etc when testing locally.

**Comment by est31** [comment]
@mitchmindtree can you look at it again?

**Review by mitchmindtree — APPROVED** [review]
Looks good to me!

**Pull request merged**